overlord/snapstate/snapstate.go

```diff
@@ -68,7 +68,7 @@ func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int
 		}
 		if model == nil || model.Base() == "" {
 			tr := config.NewTransaction(st)
-			experimentalAllowSnapd, err := getFeatureFlagBool(tr, "experimental.snapd-snap")
+			experimentalAllowSnapd, err := getFeatureFlagBool(tr, "experimental.snapd-snap", false)
```

Comment:
How about just dropping the code that looks at the flag? It should still work if you revert snapd because the actual setting is not removed.

Reply:
I had the same feeling when reviewing the code, YAGNI and all that. But we had a quick irc discussion with Gustavo and he likes the idea of keeping the flag.